### PR TITLE
Show a login required notification for the examples

### LINKED ISSUE #48: Add a friendly message to the demo page when the user is logged out
## Description

Add a friendly message to the demo page when the user is logged out. As of now, the page is showing loading states and errors are being raised in the console.

### Ideas
- Add an informative text to the examples list page (http://127.0.0.1:8000/) mentioning the user should be logged in to use the demos
  -  https://mantine.dev/core/alert/

### PR DESCRIPTION
**Pull request by pamella**
## Issues

Resolves #48

## Description

- Add an info alert about the need to log in to engage with the examples
- Update list styling

[Screencast from 27-06-2024 11:08:28.webm](https://github.com/vintasoftware/django-ai-assistant/assets/22326737/a0923a1c-ca4e-4d4d-980d-ed37a73843b7)

### CODE REVIEW

File: example/assets/js/App.tsx [hunk 1]
```diff
@@ -19,22 +29,69 @@ const theme = createTheme({});
 configAIAssistant({ BASE: "ai-assistant" });
 
 const ExampleIndex = () => {
+  const [showAlert, setShowAlert] = React.useState<boolean>(true);
```
Comment (fjsj) [inline example/assets/js/App.tsx:32]: Can we only show that if the user is logged out?

Comment (pamella) [reply example/assets/js/App.tsx:32]: Hm, a workaround would be to do a simple request to the API, using an existing endpoint, and then check if the response status is 401. Would that work for the examples? I can add a code comment saying that in real projects it should do a request related to the User entity.

Comment (fjsj) [reply example/assets/js/App.tsx:32]: Yes, that's fine

File: example/assets/js/App.tsx [hunk 2]
```diff
@@ -19,22 +33,87 @@ const theme = createTheme({});
...
 
 const ExampleIndex = () => {
+  const [showAlert, setShowAlert] = React.useState<boolean>(false);
+
+  const { fetchAssistants } = useAssistantList();
+
+  useEffect(() => {
+    // NOTE: In a real application, you should use a more robust
+    // authentication check strategy than this.
+    async function checkUserAuthenticated() {
```
Comment (fjsj) [inline example/assets/js/App.tsx:43]: checkUserAuthenticated -> checkUserIsAuthenticated

File: example/assets/js/App.tsx [hunk 3]
```diff
@@ -19,22 +33,87 @@ const theme = createTheme({});
 configAIAssistant({ BASE: "ai-assistant" });
 
 const ExampleIndex = () => {
+  const [showAlert, setShowAlert] = React.useState<boolean>(false);
```
Comment (fjsj) [inline example/assets/js/App.tsx:36]: showAlert -> showLoginAlert

**Review by fjsj — DISMISSED** [review]

**Comment by pamella** [comment]
@fjsj Here is the up-to-date behavior:

[Screencast from 27-06-2024 12:11:19.webm](https://github.com/vintasoftware/django-ai-assistant/assets/22326737/5ed84c54-4d7e-45fb-812d-306702fdd866)

**Review by fjsj — APPROVED** [review]

**Pull request merged**